internal/controller/dragonfly_instance.go

```diff
@@ -87,7 +87,7 @@ func (dfi *DragonflyInstance) configureReplication(ctx context.Context) error {
 	var master string
 	var masterIp string
 	for _, pod := range pods.Items {
-		if pod.Status.Phase == corev1.PodRunning && pod.Status.ContainerStatuses[0].Ready && pod.Labels[resources.Role] != resources.Master && pod.DeletionTimestamp == nil {
+		if pod.Status.Phase == corev1.PodRunning && pod.Status.ContainerStatuses[0].Ready && pod.Labels[resources.Role] != resources.Master && pod.DeletionTimestamp == nil && pod.Status.PodIP != "" {
```

Comment:
Can we have Status.ContainerStatuses[0].Ready while the IP was not set?

If so doesn't it mean we need to change our readiness probe?

Reply:
Our readiness (& healthcheck) probes are both the same and check if the PING commands in the container work. Pod IP assignment happens from K8s and can happen before or after. I've seen some cases where the IP was empty and we were trying to connect, and hence the additional check.